src/components/service-worker-registration/index.tsx

```diff
@@ -0,0 +1,23 @@
+'use client';
+import { useEffect } from 'react';
+
+export default function ServiceWorkerRegistration() {
+  useEffect(() => {
+    if ('serviceWorker' in navigator) {
+      window.addEventListener('load', () => {
+        navigator.serviceWorker.register('/sw.js').then(
+          (registration) => {
+            console.log(
+              'Service Worker registration successful with scope: ',
+              registration.scope,
+            );
+          },
+          (err) => {
+            console.log('Service Worker registration failed: ', err);
+          },
+        );
+      });
+    }
+  }, []);
+  return null;
+}
```

Comment:
We can use the package for this implementation for better support: https://serwist.pages.dev/docs/next

Reply:
Why can't we use  [Next-pwa](https://github.com/shadowwalker/next-pwa). It is more popular than the above mentioned

Reply:
I will pick up this next PR